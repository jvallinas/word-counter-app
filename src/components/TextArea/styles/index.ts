import styled from "styled-components";

export const StyledTextArea = styled.textarea`
  width: 90%;
`;
StyledTextArea.displayName = "StyledTextArea";
