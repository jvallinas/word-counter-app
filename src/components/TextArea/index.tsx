import React from "react";
import { StyledTextArea } from "./styles";

const PLACEHOLDER_TEXT = "Enter some text here";

interface Props {
  currentInput: string;
  onChangeHandler: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  cols?: number;
  rows?: number;
  placeholder?: string;
}

export const TextArea: React.FC<Props> = ({
  currentInput,
  onChangeHandler,
  cols,
  rows,
  placeholder = PLACEHOLDER_TEXT,
}) => {
  return (
    <>
      <StyledTextArea
        value={currentInput}
        onChange={onChangeHandler}
        placeholder={placeholder}
        cols={cols}
        rows={rows}
      ></StyledTextArea>
    </>
  );
};

export default TextArea;
