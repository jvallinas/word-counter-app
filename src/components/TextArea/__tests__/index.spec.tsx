import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TextArea from "..";

configure({ adapter: new Adapter() });

describe("TextArea component", () => {
  it("should render properly", () => {
    const component = shallow(
      <TextArea currentInput={"mockText"} onChangeHandler={() => {}} />
    );

    expect(component).toMatchSnapshot();
  });
});
