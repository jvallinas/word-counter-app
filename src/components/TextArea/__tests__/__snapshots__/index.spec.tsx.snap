// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component should render properly 1`] = `
<Fragment>
  <StyledTextArea
    onChange={[Function]}
    placeholder="Enter some text here"
    value="mockText"
  />
</Fragment>
`;
